svgelements/svgelements.py

```diff
@@ -7333,19 +7333,30 @@ def index_to_path_index(self, index):
             return slice(start, stop, step)
         return self._numeric_index(index)
 
-    def bbox(self):
+    def bbox(self, transformed=True, with_stroke=False):
         """returns a bounding box for the input Path"""
-        segments = self._path._segments[self._start : self._end + 1]
+        segments = self._path.segments(transformed=transformed)
```

Comment:
This is a bit expensive. Calling actual segments() performs this transformation on the entire list of segments when only a subsection of the segments actually matters. If you are performing a minor `bbox` operations on this subpath it would be fine. But, here you're doing the entire backing path in order to extract the couple subsections you care about.

In this case it might be better to do Path(subpath).bbox() if that's the desire there. Subpath isn't a shape after all, it's a minor hollow window class into a path data.

Reply:
In fact, the easier code here is merely: `return Path(self).bbox(transformed=transformed, with_strokes=with_strokes)` while this wraps the code and makes a copy it's actually going to do the same operations as a fancier bit of code.

The reason bbox() exists for path is that there was a minor operation that could be done without duplicating the segments, but this precluded transformations which would absolutely require that.

Reply:
If you specify with transformation = true, it should have to create the Path() copy and call the Path.bbox(), but if the transformation=False, you can actually just lookup the subset as the current code currently does.

And all of this would need test coverage.

Reply:
> `return Path(self).bbox(transformed=transformed, with_strokes=with_strokes)`

Agreed

> And all of this would need test coverage.

I will add.